packages/yourloops/components/statistics/patient-statistics.tsx

```diff
@@ -54,17 +56,23 @@ export interface PatientStatisticsProps {
   dateFilter: DateFilter
 }
 
-export const PatientStatistics: FunctionComponent<PropsWithChildren<PatientStatisticsProps>> = (props) => {
-  const { medicalData, bgPrefs, bgType, dateFilter, children } = props
+export const PatientStatistics: FunctionComponent<PatientStatisticsProps> = (props) => {
+  const { medicalData, bgPrefs, bgType, dateFilter } = props
   const theme = useTheme()
   const location = useLocation()
-
   const cbgSelected = bgType === DatumType.Cbg
   const cbgStatType: CBGStatType = cbgSelected ? CBGStatType.TimeInRange : CBGStatType.ReadingsInRange
   const numberOfDays = TimeService.getNumberOfDays(dateFilter.start, dateFilter.end, dateFilter.weekDays)
   const bgUnits = bgPrefs.bgUnits
   const selectedBgData = cbgSelected ? medicalData.cbg : medicalData.smbg
   const isTrendsPage = location.pathname.includes('trends')
+  const manufacturer = medicalData.pumpSettings.map(item => item.payload.device.manufacturer)
+  const deviceModel = medicalData.pumpSettings.map(item => item.payload.device.name)
+
+  const isAutomatedBasalDevice = (): boolean => {
+    const models = automatedBasalDeviceModel[checkManufacturerName(manufacturer)] || false
+    return Array.isArray(models) && models.includes(deviceModel)
+  }
```

Comment:
This function doesn't work. In my test the patient has a device name `DBLG1` which is our automated device. But it's not present in the `deviceModel` variable, so it will never be displayed. You need to rethink how this function works because it's not explicit !